### PR TITLE
Fix: Now specifies fast-safe-stringify proper version (1.1.x)

### LINKED ISSUE #21: patch for fast-safe-stringify dependency
This issue https://github.com/davidmarkclements/fast-safe-stringify/issues/4 makes the latest release unsafe to use. Would you consider releasing a patch `4.0.1` with the updated dependency?

### PR DESCRIPTION
**Pull request by glesperance**
Fixes #21 
(for  "caret ranges" see https://github.com/npm/node-semver#caret-ranges-123-025-004)

### CODE REVIEW

**Comment by Marsup** [comment]
Unlikely to be merged if you use caret, go for `1.1.x`.

**Comment by glesperance** [comment]
@Marsup Could you go a bit more into detail as to why? It caret ranges seem to be the default for npm/yarn.

Also given the assumption the deps are using semver, it seems that caret is the best option.

**Comment by glesperance** [comment]
Also, If I put  "1.x.x" (just like `"hoek": "4.x.x"`) this is effectively the same as going for caret ranges.

**Comment by Marsup** [comment]
I said `1.1.x`. This org doesn't rely on major semver for external modules, only minor, `hoek` is internal so it's on a major.

**Comment by glesperance** [comment]
@Marsup Cool cool cool. 
Juste updated the PR branch.

For future reference can you point me to a style guide?

**Comment by Marsup** [comment]
I was not saying this harshly fwiw, emotions don't convey well when in a hurry, sorry for that ;)

This has been a tacit rule since the birth of the project, don't think we ever wrote that though. @nlf, saw it somewhere ?

**Comment by glesperance** [comment]
@Marsup no worries / no offence was taken :)

File: package.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
     "node": ">=4.0.0"
   },
   "dependencies": {
-    "fast-safe-stringify": "1.1.0",
+    "fast-safe-stringify": "1.1.x",
```
Comment (arb) [inline package.json]: As a rule, we don't use `x` for dependencies "we" don't control. Please set it to the version with the known fix.

Comment (Marsup) [inline package.json]: Uh ? Yes we do, just on patch.

Comment (arb) [inline package.json]: Nope. That's a rule passed down from Eran.

Comment (glesperance) [inline package.json]: This is, unfortunately not possible at the moment, as I am also waiting on fast-safe-stringify to accept a PR for another bug. I'll update the PR when possible.

**On a more general note:**

I understand you guys want to have control on how your package behaves but if you don't trust the maintainer of your deps to at least respect rev versioning perhaps you're using the wrong deps.

I'd like to argue that this goes against the spirit of NPM & open source by stating that if everyone fixed all of their deps to a specific version we'd have to do a PR to each package up the chain of dependencies each time a bug is found...

Not only is this considerably less productive than having trust in our deps ; but it also casts doubts about how fast bugs can be fixed in good-squeeze.

For instance: at the moment, good-squeeze crashes node in unexpected ways since it is using a version of fast-safe-stringify that is 11 revisions behind...

In other words, it'd be much appreciated if you reconsidered your stance on this.

Comment (glesperance) [inline package.json]: Just to add to this ; relying on specific versions of fast-safe-stringify will require you not only to review PRs for each of their releases but as well to do a release each time they do...

This doubles the work, but also the time to fix a problem

Comment (arb) [inline package.json]: Since the people behind "fast-safe-stringify" seem to take semver serious, I'll take this as is.

Comment (glesperance) [inline package.json]: Awesome ; thank you!
